src/main/scala/com/datastax/gatling/plugin/utils/CqlPreparedStatementUtil.scala

```diff
@@ -44,12 +45,80 @@ trait CqlPreparedStatementUtil {
   def getParamsList(preparedStatement: PreparedStatement): List[Int]
 }
 
+object SessionCollectionResolver {
+  def get(session:Session, name:String):Option[Any] = session.attributes.get(name)
+
+  def getClz[T <: Any](session:Session, name:String):Option[Class[T]] = {
+    get(session,name).flatMap((sessionVal) => {
+      sessionVal match {
+        case clz:Class[T] => Option(clz)
+        case _ => Option.empty
+      }
+    })
+  }
+
+  def getIterable[T <: Any](session:Session, name:String):Option[lang.Iterable[T]] = {
+    get(session,name).flatMap((sessionVal) => {
+      sessionVal match {
+        case rv:Iterable[T] => Option(rv.asJava)
+        case rv:lang.Iterable[T] => Option(rv)
+        case _ => Option.empty
+      }
+    })
+  }
+
+  def getMap[K <: Any, V <: Any](session:Session, name:String):Option[util.Map[K,V]] = {
+    get(session,name).flatMap((sessionVal) => {
+      sessionVal match {
+        case rv:Map[K,V] => Option(rv.asJava)
+        case rv:util.Map[K,V] => Option(rv)
+        case _ => Option.empty
+      }
+    })
+  }
+
+  def getIterableClz[T <: Any](session:Session, name:String):Class[_ <: T] = {
+    val sessionOption:Option[Class[T]] = getClz(session, name + "-clz")
+    if (sessionOption.isDefined) {
+      sessionOption.get
+    } else {
+      val iterableOption:Option[lang.Iterable[T]] = getIterable(session, name)
```

Comment:
if the conversion will be to scala iterators, then we will not need to use `lang.Iterable[T]` but just scala `Iterator[T]`

Reply:
Agreed, will be addressed in changes referenced above.